core-client/src/lib.rs

```diff
@@ -597,6 +597,14 @@ pub struct RecoveryParameters {
     pub custodian_recovery_outputs: Vec<PathBuf>,
 }
 
+#[derive(Debug, Parser, Clone)]
+pub struct ReshareParameters {
+    // ID of the key to reshare
+    pub key_id: RequestId,
+    // ID of the preprocessing used to generate the key
+    pub preproc_id: RequestId,
```

Comment:
Why is the preprocessing ID needed? Presumably you mean the ID used when originally generating the key?

Reply:
Right, let me rephrase; why should it be known by the caller at this time? Once a key has been generated I don't think the caller should keep being aware of the preproc ID used for the original key's generation. It seems it is rather metadata the KMS server should keep track of.